src/npg_porch_cli/api.py

```diff
@@ -55,100 +48,80 @@ class ServerErrorException(Exception):
     pass
 
 
-@dataclass
-class PorchRequest:
+@dataclass(kw_only=True)
+class Pipeline:
+
+    name: str
+    uri: str
+    version: str
+
+    def __post_init__(self):
```

Comment:
Where did this idea come from? I'd like to read around it a bit. Naïvely I assumed it would be an `__init__` with the three required properties as arguments, although you still could perversely pass `None` as a value. The RealPython document on using properties looks positively clumsy: https://realpython.com/python-property/#validating-input-values

Reply:
https://docs.python.org/3/library/dataclasses.html#post-init-processing Though the idea to validate like this did not have to come from anywhere. If I have to go down the route of validating individual fields, I'd move to pydantic, which I did not want to have as a dependency